### ISSUE TITLE
Upload Progress

### ISSUE THREAD

**Issue by MarvinNazari** [ishkawa/APIKit#186]
Sorry if i ask too many question here, wonder if there is any way we could get a progress when we upload a file like Alamofire upload method. Thanks 👍

**Comment by MarvinNazari** [comment]
Is there any news on this?

**Comment by ishkawa** [comment]
For now, APIKit does not provide way to get progress of uploading. I'm planning to add a parameter to `Session.sendRequest()` for this purpose. The call-site will look like:

```swift
Session.sendRequest(request, 
    progressHandler: { bytesSent, totalBytesSent, totalBytesExpectedToSend in
        ...
    },
    handler: { result in
        ...
    })
```

**Comment by ishkawa** [comment]
> Sorry if i ask too many question here

Any questions are welcome 👍
I think your questions help other developers and inspire me.

**Comment by MarvinNazari** [comment]
ah thats awesome, we are really in hurry! is there anyway i can help with this?

**Comment by klaus01** [comment]
Version `3.1.1` has no `sendRequest (request: progressHandler: handler :)` method.

How can we achieve the upload progress?

**Comment by MarvinNazari** [comment]
@ishkawa: any progress on this?

**Comment by taish** [comment]
I really want it too!

**Comment by taish** [comment]
@ishkawa: I implement your nice idea :wink:
please review it

**Comment by ikesyo** [comment]
Closed by #234.
